hngx-org/authentication_service
# feat/reset password

# Forgot Password and Reset Feature

## Flow
1. User requests for a password reset link from the frontend. They have to include their email.
2. The link sends a POST request to the endpoint `/api/auth/reset-password` with the email in the body `{"email": "example@mail.com"}`
3. The endpoint then generates a token, and sends it to the user's email. This service communicates with the Messaging service to facilitate this.
4. The email link sends the user to the endpont `/api/auth/reset-password/:token`. This is a `GET` request.
5. The endpoint verifies the validity of the token and redirects the user to the frontend password reset page. And passes the token as a param.
6. The frontend dev passes the validated password to the endpoint `/api/auth/reset-password` this is a `PATCH` request. with the token and email in the body of the request.
7. Upon verification of the token once more, the password is updated.


## Commits Associated with the merge.
- import sendverificationemail helper
- clean up index.js
- create index router and serve swagger ui
- set up swagger controller
- check-email route
- refactor signup endpoint
- verify user endpoint
- verify email endpoint and redirection
- resnd verification endpoint
- reset password endpiont, controller and messaging service
- request password reset link endpiont
- reset password flow complete
